### PR TITLE
protected structures fix: they were made public + Time synchronization classes were added

### PR DESCRIPTION
**Pull request by alexkernphysiker**
there was compile error in dictionary files

### CODE REVIEW

**Comment by wkrzemien** [comment]
You also added LargeBarrelExtensions  with this PR - do you want to add now? - then change the title pls -?

File: LargeBarrelExtensions/BarrelExtensions.cpp [hunk 1]
```diff
@@ -0,0 +1,141 @@
+#include <algorithm>
+#include <sstream>
+#include "BarrelExtensions.h"
+using namespace std;
+const string Layer(const size_t layer){
+    stringstream stream;
```
Comment (wkrzemien) [inline LargeBarrelExtensions/BarrelExtensions.cpp]: In  c++11 there is to_string method in the standard library, so you should use it

(By the way we also should do it in other part of the framework, we had some tools in JPetCommonTools, now it becomes obsolete)

File: LargeBarrelExtensions/BarrelExtensions.cpp [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    return stream.str();
+}
+const string LayerSlot(const size_t layer, const size_t slot){
+    stringstream stream;
+    stream<<"layer"<<layer<<"-slot"<<slot;
+    return stream.str();
+}
+const string LayerSlotThr(const size_t layer, const size_t slot, const size_t thr){
+    stringstream stream;
+    stream<<"layer"<<layer<<"-slot"<<slot<<"-thr"<<thr;
```
Comment (wkrzemien) [inline LargeBarrelExtensions/BarrelExtensions.cpp]: the same as above

File: LargeBarrelExtensions/BarrelExtensions.h [hunk 3]
```diff
@@ -0,0 +1,76 @@
+#ifndef _LARGE_BARREL_EXTENSIONS_
```
Comment (wkrzemien) [inline LargeBarrelExtensions/BarrelExtensions.h]: licence header is missing

**Comment by alexkernphysiker** [comment]
I first made a pull request, than I committed new changes to my repository. what am I doing wrong?

File: LargeBarrelExtensions/BarrelExtensions.cpp [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    return {.layer=getLayerNumber(slot.getLayer()),.slot=getSlotNumber(slot)};
+}
+const vector< size_t > AbstractBarrelMapping::getLayersSizes() const{
+    vector<size_t> res;
+    for(size_t i=1,n=getLayersCount();i<=n;i++)
+	res.push_back(getSlotsCount(i));
+    return res;
+}
+
+const double degFullCircle = 360.;
```
Comment (wkrzemien) [inline LargeBarrelExtensions/BarrelExtensions.cpp]: Do you need this global?
btw we have a convention to start consts with kDegFullCircle

Comment (alexkernphysiker) [inline LargeBarrelExtensions/BarrelExtensions.cpp]: It was just taken from Alek's example. OK, I'll change it

File: LargeBarrelExtensions/BarrelExtensions.cpp (hunk 1 above)
Comment (wkrzemien) [inline LargeBarrelExtensions/BarrelExtensions.cpp]: It is shorter and more expressive:
return "layer" + to_string(layer);

**Comment by wkrzemien** [comment]
If  you add new changes to the same branch on your github account it automatically updates PR. you must create a separate branch if you want to add some new stuff and not affect the PR

File: LargeBarrelExtensions/BarrelExtensions.cpp (hunk 1 above)
Comment (alexkernphysiker) [inline LargeBarrelExtensions/BarrelExtensions.cpp]: OK. I'll yield to the dictation. Nevertheless, I don't think that this one is less expressive

File: LargeBarrelExtensions/BarrelExtensions.cpp (hunk 4 above)
Comment (alexkernphysiker) [inline LargeBarrelExtensions/BarrelExtensions.cpp]: :laughing:  It even wasn't used. Deleting!

**Comment by wkrzemien** [comment]
-You should change methods names to small letters (now it is a mixture of both cases - sometimes small, sometimes capital).
-Add some description to Synchronization class.
-And there is some conflict in CMakeLists.txt

**Comment by alexkernphysiker** [comment]
Which conflict do you mean?

**Pull request merged**